### PR TITLE
Remove unnecessary era parameters and use ShelleyBasedEra era

### PR DESCRIPTION
**Pull request by Jimbo4350**
# Changelog

```yaml
- description: |
    Remove unnecessary era parameters and replace `CardanoEra era` with `ShelleyBasedEra era` where appropriate
  type:
  - refactoring   
```

# Context

Additional context for the PR goes here. If the PR fixes a particular issue please provide a [link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=) to the issue.

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

**Review by carbolymer — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/EraBased/Commands.hs [hunk 1]
```diff
@@ -114,17 +114,16 @@ pAnyEraCommand envCli =
 
 pCmds :: ShelleyBasedEra era -> EnvCli -> Parser (Cmds era)
 pCmds sbe' envCli = do
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/EraBased/Commands.hs]:
```suggestion
pCmds sbe envCli = do
```

Take the occasion to rename to a more standard name?

File: cardano-cli/src/Cardano/CLI/EraBased/Options/StakeAddress.hs [hunk 2]
```diff
@@ -91,74 +83,70 @@ pStakeAddressBuildCmd era envCli = do
...
-                  <$> pStakeIdentifier Nothing
-                  <*> fmap Just pKeyRegistDeposit
-                  <*> pOutputFile
-              )
-            $ Opt.progDesc "Create a stake address deregistration certificate"
-      )
-      sbe
+  => ShelleyBasedEra era
+  -> Parser (StakeAddressCmds era)
+pStakeAddressDeregistrationCertificateCmd sbe = do
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/StakeAddress.hs]: Do you need the `do` here?

Comment (smelc) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/StakeAddress.hs]: You can also probably eta reduce this function.

**Review by smelc — APPROVED** [review]
Nice simplification :+1:

**Pull request merged**